### PR TITLE
General - Fix some undefined variables

### PR DESCRIPTION
**Pull request by PabstMirror**

### CODE REVIEW

File: addons/common/XEH_postInit.sqf [hunk 1]
```diff
@@ -197,7 +197,7 @@ if (isServer) then {
 
 [QGVAR(unlockVehicle), {
     _this lock (_this getVariable [QGVAR(lockStatus), locked _this]);
-    if ([] isNotEqualTo getArray (configOf _target >> "assembleInfo" >> "dissasembleTo")) then {
+    if ([] isNotEqualTo getArray (configOf _this >> "assembleInfo" >> "dissasembleTo")) then { // double check
```
Comment (PabstMirror) [inline addons/common/XEH_postInit.sqf:200]:
```suggestion
    if ([] isNotEqualTo getArray (configOf _this >> "assembleInfo" >> "dissasembleTo")) then {
```

File: addons/common/functions/fnc_checkFiles.sqf [hunk 2]
```diff
@@ -144,7 +144,7 @@ if (isMultiplayer) then {
         publicVariable QGVAR(serverAddons);
         publicVariable QGVAR(serverSource);
     } else {
-        GVAR(clientVersion) = _version;
+        GVAR(clientVersion) = _mainVersion; // double check
```
Comment (PabstMirror) [inline addons/common/functions/fnc_checkFiles.sqf:147]:
```suggestion
        GVAR(clientVersion) = _mainVersion;
```

File: addons/cookoff/functions/fnc_cookOffServer.sqf [hunk 3]
```diff
@@ -199,4 +199,4 @@ if (_delayBetweenSmokeAndFire) then {
 }, [_vehicle, _selections, _ammoDetonationChance, _detonateAfterCookoff, _source, _instigator, _fireSelection, _canRing, _canJet, _smokeJipID, _fireJipID], _delay] call CBA_fnc_waitAndExecute;
 
 // API
-[QGVAR(cookoff), [_vehicle, _intensity, _instigator, _smokeDelayEnabled, _ammoDetonationChance, _detonateAfterCookoff, _fireSelection, _canRing, _maxIntensity, _canJet]] call CBA_fnc_globalEvent;
+[QGVAR(cookoff), [_vehicle, _intensity, _instigator, _delayBetweenSmokeAndFire, _ammoDetonationChance, _detonateAfterCookoff, _fireSelection, _canRing, _maxIntensity, _canJet]] call CBA_fnc_globalEvent; // double check
```
Comment (PabstMirror) [inline addons/cookoff/functions/fnc_cookOffServer.sqf:202]:
```suggestion
[QGVAR(cookoff), [_vehicle, _intensity, _instigator, _delayBetweenSmokeAndFire, _ammoDetonationChance, _detonateAfterCookoff, _fireSelection, _canRing, _maxIntensity, _canJet]] call CBA_fnc_globalEvent;
```

File: addons/grenades/functions/fnc_flashbangExplosionEH.sqf [hunk 4]
```diff
@@ -113,7 +113,7 @@ if (["ace_hearing"] call EFUNC(common,isModLoaded) && {_strength > 0} && {EGVAR(
 };
 
 // Add ace_medical pain effect
-if (GETEGVAR(medical,enabled,false) && {_strength > 0.1} && {isDamageAllowed _unit} && {_unit getVariable [QEGVAR(medical,allowDamage), true]}) then {
+if (GETEGVAR(medical,enabled,false) && {_strength > 0.1} && {isDamageAllowed ACE_player} && {ACE_player getVariable [QEGVAR(medical,allowDamage), true]}) then { // double check
```
Comment (PabstMirror) [inline addons/grenades/functions/fnc_flashbangExplosionEH.sqf:116]:
```suggestion
if (GETEGVAR(medical,enabled,false) && {_strength > 0.1} && {isDamageAllowed ACE_player} && {ACE_player getVariable [QEGVAR(medical,allowDamage), true]}) then {
```

File: addons/grenades/functions/fnc_incendiary.sqf [hunk 5]
```diff
@@ -202,7 +202,7 @@ if (isServer) then {
             {EGVAR(cookoff,ammoCookoffDuration) != 0} &&
             {_x getVariable [QEGVAR(cookoff,enableAmmoCookoff), true]}
         ) then {
-            [QEGVAR(cookOff,cookOffBoxServer), _box] call CBA_fnc_serverEvent;
+            [QEGVAR(cookOff,cookOffBoxServer), _x] call CBA_fnc_serverEvent; // double check
```
Comment (PabstMirror) [inline addons/grenades/functions/fnc_incendiary.sqf:205]:
```suggestion
            [QEGVAR(cookOff,cookOffBoxServer), _x] call CBA_fnc_serverEvent;
```

File: addons/common/functions/fnc_checkFiles.sqf (hunk 2 above)
Comment (PabstMirror) [reply addons/common/functions/fnc_checkFiles.sqf:147]: it's work in progress right now

**Comment by PabstMirror** [comment]
I know all the `//IGNORE` stuff is a little obnoxious, but I think it's worth it to help find errors

**Comment by PabstMirror** [comment]
Should I rip out the real changes to other prs and leave the annotations separate?

File: addons/repair/functions/fnc_patchRemovedWheel.sqf [hunk 6]
```diff
@@ -23,7 +23,7 @@ params ["_unit", "_target"];
 private _wheelDamage = (damage _target) - GVAR(patchWheelMaximumRepair);
 
 [ceil (_wheelDamage / 0.05) * GVAR(patchWheelTime), [_target], {}, {}, LLSTRING(PatchingWheel), {
-    params ["_args"];
+    params ["_args", "_elapsedTime", "_totalTime"];
```
Comment (johnb432) [inline addons/repair/functions/fnc_patchRemovedWheel.sqf:26]: Reminder to self: Update line 38 to use `EFUNC`.

File: addons/javelin/functions/fnc_showFireMode.sqf [hunk 7]
```diff
@@ -14,11 +14,12 @@
  *
  * Public: No
  */
+//IGNORE_PRIVATE_WARNING ["_currentShooter"]; // from upper scope
 
 (if (((vehicle ACE_player) == ACE_player) || {ACE_player call CBA_fnc_canUseWeapon}) then {
     weaponState ACE_player
 } else {
-    _turretPath = if (ACE_player == (driver _currentShooter)) then {[-1]} else {ACE_player call CBA_fnc_turretPath};
+    private _turretPath = if (ACE_player == (driver _currentShooter)) then {[-1]} else {ACE_player call CBA_fnc_turretPath};
```
[the remark this answers is not in the document]
Comment (PabstMirror) [reply addons/javelin/functions/fnc_showFireMode.sqf:22]: `CBA_fnc_turretPath` handles driver turret as []`
```
// driver will be returned as [] to be consistent with previous behaviour
if (_turret isEqualTo [-1]) exitWith { [] };
```

don't think you would ever have a driver javelin? but I don't think this is very performance intensive

[the remark this answers is not in the document]
Comment (johnb432) [reply addons/javelin/functions/fnc_showFireMode.sqf:22]: No it's not performance intensive at all, just wondering if we could make it a one liner - doesn't seem to be the case though.

**Comment by johnb432** [comment]
> Should I rip out the real changes to other prs and leave the annotations separate?

Imo no.

**Review by johnb432 — APPROVED** [review]

**Pull request merged**